sentry_sdk/integrations/starlette.py

```diff
@@ -362,13 +363,19 @@ def patch_middlewares():
 
     if not_yet_patched:
 
-        def _sentry_middleware_init(self, cls, **options):
-            # type: (Any, Any, Any) -> None
+        def _sentry_middleware_init(self, cls, *args, **kwargs):
+            # type: (Any, Any, Any, Any) -> None
             if cls == SentryAsgiMiddleware:
-                return old_middleware_init(self, cls, **options)
+                if StarletteIntegration.starlette_version < (0, 35):
+                    return old_middleware_init(self, cls, **kwargs)
+                else:
+                    return old_middleware_init(self, cls, *args, **kwargs)
```

Comment:
[suggestion]

Why not just always pass the `*args`, like so?

```suggestion
                return old_middleware_init(self, cls, *args, **kwargs)
```

I am guessing the ability to pass positional arguments was only added in starlet 0.35, but in that case, users would simply omit positional arguments on that version. My guess is they'd probably get an error from Starlette directly if they passed positional arguments (which would be a good thing in this case, since they would know they are using something their Starlette version lacks support for).

If anything, I think this check could be harmful, since users who use the SDK would not get any error if they passed positional arguments to the middleware. These would just be silently ignored, and the user would not know why their arguments are not being honored.

Or, am I missing something here?

Reply:
Good reasoning. I changed it! (On very old Starlette if someone adds a positional args they will get an error and it will seem like it is a problem with Sentry (because the stack trace will have a sentry function in it, but I guess almost noone is still using this very early versions of Starlette.)